.github/workflows/ui-build.yml

```diff
@@ -0,0 +1,29 @@
+name: UI Build
+
+on:
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'src/ui/**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "16"
+
+      - name: Install dependencies
+        working-directory: ./src/ui/app
+        run: npm install
+
+      - name: Format check
+        working-directory: ./src/ui/app
+        run: npm run lint
+
+      - name: Build
```

Comment:
Rather than this, let's do `make link-dev` and then run `make dist`. That should cover everything we need to do.

Reply:
those commands don't call `npm install` in the `app/` directory though. Was that an oversight or is that ok?

Reply:
`npm install` is actually unnecessary. If `npm run build` in the `app` directory succeeds, that means that the code has compiled and built into a bundle that we can use.

Reply:
Woah. That's super weird. It's not installing force-resolutions, but it's still working.